apps/server/src/controllers/userControllers/authUsers.ts

```diff
@@ -360,3 +362,68 @@ export const resetPassword = async (req: Request, res: Response) => {
     })
   }
 }
+
+export const changePassword = async (req: RequestExt, res: Response) => {
+  const passwordRegex = passwordUtils.regex
+
+  try {
+    const { _userId: userId, ...rest } = req.body
+
+    const requestData = changePasswordSchema.strict().safeParse(rest)
+
+    if (!requestData.success) {
+      return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+        message: requestData.error.issues,
+      })
+    }
+
+    const _data = requestData.data
+
+    if (!passwordRegex.test(_data.newPassword)) {
+      return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+        message: passwordUtils.error,
+      })
+    }
+
+    const user = await Users.findOne({
+      where: { id: userId },
+    })
+
+    // Check if the user exists
+    if (!user) {
+      return res.status(HTTP_STATUS_CODE.NOT_FOUND).json({
+        message: 'User not found',
+      })
+    }
+
+    // check if the current password is correct
+    const confirmPassword = await PasswordHarsher.compare(
+      _data.password,
+      user.password
+    )
+
+    if (!confirmPassword)
+      return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+        message: 'Wrong password',
+      })
```

Comment:
This should be a 401

because it's an unauthorized request 

it's not a bad request if what the server is asking for was passed correctly in the first place

Reply:
I thought this error will fire if the user enters his oldPassword incorrectly making it a bad request?

Reply:
Yeah, it should be an unauthorized status code.